dev-python/nose/nose-1.3.7.recipe

```diff
@@ -0,0 +1,109 @@
+working directory whose names include "test" or "Test" at a word boundary \
+(like "test_this" or "functional_test" or "TestClass" but not \
+"libtest"). Test output is similar to that of unittest, but also includes \
+captured stdout output from failing tests, for easy print-style debugging.
+
+These features, and many more, are customizable through the use of plugins. \
+Plugins included with nose provide support for doctest, code coverage, and \
+profiling, flexible attribute-based test selection, output capture and more. \
+More information about writing plugins may be found on in the nose API \
+documentation, here:
+http://readthedocs.io/docs/nose/'
+HOMEPAGE="https://readthedocs.io/docs/nose/
+	https://pypi.python.org/pypi/nose/
+	https://github.com/nose-devs/nose"
+COPYRIGHT="2008-2010 anatoly techtonik"
+LICENSE="GNU LGPL v2.1"
+REVISION="1"
+SOURCE_URI="https://github.com/nose-devs/nose/archive/release_$portVersion.tar.gz"
+CHECKSUM_SHA256="ab1ee7744662c5bf58a510ae46d879ebe140c3e40d6871c09b36b24b6a480b39"
+SOURCE_FILENAME="nose-$portVersion.tar.gz"
+SOURCE_DIR="nose-release_$portVersion"
+
+ARCHITECTURES="any"
+
+PROVIDES="
+	nose = $portVersion
+	"
+REQUIRES="
+	haiku
+	"
+BUILD_REQUIRES="
+	haiku_devel
+	"
+
+PYTHON_PACKAGES=(python python3)
+PYTHON_VERSIONS=(2.7 3.6)
+for i in "${!PYTHON_PACKAGES[@]}"; do
+pythonPackage=${PYTHON_PACKAGES[i]}
+pythonVersion=${PYTHON_VERSIONS[$i]}
+eval "PROVIDES_${pythonPackage}=\"\
+	nose_$pythonPackage = $portVersion\n\
+	cmd:nosetests_$pythonVersion
+	\"; \
+REQUIRES_$pythonPackage=\"\
+	haiku\n\
+	cmd:python$pythonVersion\
+	\""
+BUILD_REQUIRES="$BUILD_REQUIRES
+	setuptools_$pythonPackage"
+BUILD_PREREQUIRES="$BUILD_PREREQUIRES
+	cmd:python$pythonVersion"
+done
+
+BUILD()
+{
+	for i in "${!PYTHON_PACKAGES[@]}"; do
+		pythonPackage=${PYTHON_PACKAGES[i]}
+		pythonVersion=${PYTHON_VERSIONS[$i]}
+
+		rm -rf "$sourceDir"-$pythonPackage
+		cp -a "$sourceDir" "$sourceDir"-$pythonPackage
+		cd "$sourceDir"-$pythonPackage
+
+		python=python$pythonVersion
+		$python setup.py build
+	done
+}
+
+INSTALL()
+{
+	for i in "${!PYTHON_PACKAGES[@]}"; do
+		pythonPackage=${PYTHON_PACKAGES[i]}
+		pythonVersion=${PYTHON_VERSIONS[$i]}
+
+		cd "$sourceDir"-$pythonPackage
+
+		python=python$pythonVersion
+		installLocation=$prefix/lib/$python/vendor-packages
+		export PYTHONPATH=$installLocation
+		mkdir -p "$installLocation" "$docDir"
+		sed -i "s|'man/|'documentation/man/|" setup.py
+		$python setup.py install \
+			--root=/ --prefix="$prefix" --skip-build
+
+		cp -rt "$docDir" \
+			README.txt DEVELOPERS.txt CHANGELOG NOTES NEWS* doc/*
+		rm -rf $binDir/nosetests $docDir/{doc_tests,Makefile}
+		packageEntries  $pythonPackage \
+			$prefix/lib/python* \
+			$binDir
+	done
+}
+
+TEST()
+{
+	for i in "${!PYTHON_PACKAGES[@]}"; do
+		pythonPackage=${PYTHON_PACKAGES[i]}
+
+		cd "$sourceDir"-$pythonPackage
+
+		# Ironically, the test seems to fail everywhere with python3
+		python setup.py test
```

Comment:
This should make the tests work with python3 :)
```diff
-		# Ironically, the test seems to fail everywhere with python3
+		python=python$pythonVersion
-		python setup.py test
+		$python test_tzinfo.py
```

Reply:
The file doesn't seem to exist.

Reply:
My mistake. It was an incorrect copy/paste from dev-python/pytz/...

Reply:
Sorry, I forgot to add a line to define `pythonVersion`:
```diff
 		pythonPackage=${PYTHON_PACKAGES[i]}
+		pythonVersion=${PYTHON_VERSIONS[$i]}
 
 		cd "$sourceDir"-$pythonPackage/pytz/tests
 
 		python=python$pythonVersion
 		$python test_tzinfo.py
```

Reply:
That doesn't really change anything, though.

Reply:
@fbrosson suggestion does work for me though, however, some tests do fail.